lib/percy/stats.rb

```diff
@@ -8,7 +9,18 @@ def initialize(host = nil, port = nil, opts = {}, max_buffer_size = 50)
       port ||= Integer(ENV.fetch('DATADOG_AGENT_PORT', ::Datadog::Statsd::DEFAULT_PORT))
       opts[:tags] ||= []
       opts[:tags] << "env:#{ENV['PERCY_ENV'] || 'development'}"
-      super(host, port, opts, max_buffer_size)
+      retry_delay = opts[:retry_delay] || 1
+      retry_count = opts[:retry_count] || 3
+      retries = 0
+
+      begin
+        super(host, port, opts, max_buffer_size)
+      rescue SocketError => e
+        raise e if retries >= retry_count
```

Comment:
I think we should make this 100% resilient and just drop stats if DNS queries fail (instead of raising an error through the containing app, which we want to avoid in this case). You could do this by setting `host` to `"localhost"` and then retrying one last time, which should always resolve correctly.